index.js

```diff
@@ -37,24 +46,45 @@ const validate = (file, options) => {
 	return options;
 };
 
+const toArray = input => Array.isArray(input) ? input : [input];
+
 exports.buffer = async (file, options) => {
 	options = validate(file, options);
 
-	const isPid = typeof file === 'number';
+	const files = toArray(file);
+	const isPid = files.map(file => (typeof file === 'number').toString());
+
+	const {stdout} = await execFileP(bin, [files, options.size, isPid], spawnOptions);
 
-	const {stdout} = await execFileP(bin, [file, options.size, isPid], spawnOptions);
+	const buffers = stdout
+		.toString('hex')
```

Comment:
It seems very inefficient to first convert to Hex, split, and then back to binary. Why not just split the binary directly?

Reply:
Mostly simplicity. I find it much easier to understand how we're processing the buffer if we treat it as a string. While I haven't measured, I agree this is pretty inefficient compared to manipulating the buffer directly. Pushing changes without the conversions.